t/mldistwatch-misc.t

```diff
@@ -631,6 +632,55 @@ subtest "updates to existing packages " => sub {
   };
 };
 
+subtest "the notorious version zero" => sub {
+  # When we've got two dists both of which provide version 0 (or version
+  # undef), we pick the winner by mtime.  I hate this, but I will write a test
+  # so that I can someday more confidently forbid anything like this.  Good
+  # grief. -- rjbs, 2024-05-11
+  my $pause = PAUSE::TestPAUSE->init_new;
+
+  my sub touch_file ($filename, $i) {
+    my $secs = sprintf '%02i', $i;
+    system("touch", '-m', '-d', "2020-01-02T03:04:${secs}Z", $filename);
+    Process::Status->assert_ok("touching $filename");
+  }
```

Comment:
Eh this is fine, but you could use utime:
```
  my sub touch_file ($filename, $i) {
    my $time = time - $i;
    utime $time, $time, $filename || die "Couldn't touch $filename: $!\n";
  }

  # First, a zero version package.
  # Then, again, but file mtime is older.  We keep the package on file.
  # Then, again, but file mtime is newer.  We switch to the new package.
  my @tests = (
    { desc => "first upload",  ago => 30, dist_v => '1.0', want => '1.0' },
    { desc => "second upload", ago => 60, dist_v => '1.1', want => '1.0' },
    { desc => "third upload",  ago => 10, dist_v => '1.2', want => '1.2' },
  );
```

...
```
      touch_file($file, $test->{ago});

```

Reply:
Using `time()` as the basis for $time is wrong, it sets up races, but this is otherwise clearly an improvement.  *I didn't know about `utime`!*  Wow.  Fixed in new commit.